### PR TITLE
Update Android NDK and gomobile

### PR DESCRIPTION
**Pull request by albertvaka**
### What does this PR do?

Bumps the Android NDK and changes the init command (it's no longer needed to pass the NDK path).

It also bumps Gradle and the Gradle plugin versions, so they are compatible with the new NDK.

I also added `retrieve_linux_go_deps`: even if we still call `inv deps`, a lot of dependencies are reused this way.

### Describe your test plan

It builds.

### CODE REVIEW

File: tasks/go.py [hunk 1]
```diff
@@ -307,9 +307,7 @@ def deps(
             print("set ANDROID_NDK_HOME to build android")
```
Comment (mx-psi) [inline tasks/go.py:307]: We are not using `ndkhome` anymore, right? Can we remove this?

Comment (albertvaka) [reply tasks/go.py:307]: I've done that on my other PR... If I remove it here it will conflict big time 😓  Let's leave it here for this PR?

Comment (mx-psi) [reply tasks/go.py:307]: Sure, let's leave it here and do it on a new PR then

**Review by mx-psi — APPROVED** [review]
LGTM, let's remove the `ndkhome` bit in a separate PR

**Pull request merged**